RFS/src/main/java/com/rfs/common/SnapshotRepo.java

```diff
@@ -0,0 +1,44 @@
+package com.rfs.common;
+
+import java.util.List;
+
+public class SnapshotRepo {
+
+    /**
+     * Defines the behavior required to surface a snapshot repo's metadata
+     */
+    public static interface Provider {    
```

Comment:
My only points would be
1. Threw me off when seeing it as I was questioning how it was any different than other interfaces I've created, then realized it was redundant
2. Will be something we need to exclude from SonarLint rules as they will call this out. In general it may be beneficial to install the SonarLint plugin through Intellij or some other IDE and see what all is called out